### PR TITLE
CF: Update caution flag content based on DEPO staging review feedback. #8082

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/8082

## Testing done
- `bundle exec rake db:seed_rules`
- Generate Preview and publish
- view in vets-website

#### To test AC#11
An institution without an `insturl` is needed.
Use SQL below to find one or set a scorecard CSV row to not have one (institutions are related to scorecard via weam CSV column `ipeds` to scorecard CSV column `unitid`)
```
select i.facility_code, i.cross
	from scorecards s, institutions i
	where s.insturl IS NULL
	and s.cross = i.cross
```

## Acceptance criteria
- [ ] see https://github.com/department-of-veterans-affairs/va.gov-team/issues/8082

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: app/models/caution_flag.rb [hunk 1]
```diff
@@ -30,14 +30,18 @@ def self.cols_to_update(cols_map_update)
   end
 
   def self.apply_update(rule, ids)
-    cols_map_update = %i[
-      title description link_text link_url
-    ]
-
     if rule.link_url == CautionFlagRule::SCHOOL_URL
-      cols_map_update.pop
+      cols_map_update = %i[
```
Comment (erluti) [inline app/models/caution_flag.rb:34]: This method is now failing the linting with these additional lines.  I recommend braking the two branches of the `if` into their own method.  Then a test could be added for each of the methods.  So this method would look something 
like this:
```
def self.apply_update(rule, ids)
    apply_update_sql = 
      if rule.link_url == CautionFlagRule::SCHOOL_URL
        school_url_update_sql(rule, ids)
      else
        basic_update_sql(rule, ids)
      end

    CautionFlag.connection.update(apply_update_sql)
  end

  def self.school_url_update_sql
    [...]
  end

  def self.basic_update_sql
    [...]
  end
```

Comment (zurbergram) [reply app/models/caution_flag.rb:34]: codeclimate linting does not equal rubocop linting within the repository, this branch passes `make ci`

Comment (zurbergram) [reply app/models/caution_flag.rb:34]: See https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/.rubocop.yml#L90

Comment (zurbergram) [reply app/models/caution_flag.rb:36]: pulled out school url specific sql to its own method

**Review by philipwritescode — APPROVED** [review]
Looks good, but it'd be nice to have some test coverage for `update_for_school_url`.
I think this would work:
```
describe '.update_for_school_url' do
    it 'populates a SQL query' do
      version = build :version, :preview
      rule = create :caution_flag_rule, :closing_settlement_rule

      expect(CautionFlag.update_for_school_url(rule, [version.id])).to eq <<-SQL
            UPDATE caution_flags SET title = caution_flag_rules.title, description = caution_flag_rules.description,
            link_text = CASE
              WHEN institutions.insturl IS NULL THEN
                caution_flag_rules.link_text || '.'
              ELSE
                caution_flag_rules.link_text
              END,
            link_url = CASE
              WHEN LEFT(LOWER(institutions.insturl), 4) != 'http' THEN
                'http://' || institutions.insturl
              ELSE
                institutions.insturl
              END
          FROM caution_flag_rules, institutions
          WHERE caution_flag_rules.id = 1
          AND caution_flags.id in ()
          AND institutions.id = caution_flags.institution_id
      SQL
    end
  end
```

**Pull request merged**